model/fn-execution/src/main/proto/beam_fn_api.proto

```diff
@@ -193,6 +193,9 @@ message BundleApplication {
   // value represents a lower bound on the timestamps of elements that
   // are produced by this PTransform into each of its output PCollections
   // when invoked with this application.
+  //
+  // If there is no watermark reported from RestrictionTracker, the sdk should
+  // report MIN_TIMESTAMP by default.
```

Comment:
Are we sure we want to make this required? (In particular, when a watermark is provided, it may make sense to check that it is >= the input watermark for non-late data, which would be violated if we provide MIN_TIMESTAMP as a default value.

Reply:
It depends on how runner treats MIN_TIMESTAMP. On the other hand, it's also possible that user implementation returns MIN_TIMESTAMP right?

Reply:
Correct, if we fill this in we can't distinguish between the user setting MIN_TIMESTAMP and the framework defaulting to it at any point downstream. Better to let the "unset" propagate all the way tot the runner.

Reply:
Changed in commit https://github.com/apache/beam/pull/9023/commits/127ab262ae26b0906bd009875a01b6ea232b06ab